File: rocket-bi-web/di_core/DataCook/Domain/ETL/ThirdPartyPersistConfiguration/JdbcPersistConfiguration.ts
import { PERSISTENT_TYPE } from '@core/DataCook/Domain';
import { ThirdPartyPersistConfiguration } from '@core/DataCook/Domain/ETL/ThirdPartyPersistConfiguration/ThirdPartyPersistConfiguration';
import { ThirdPartyPersistConfigurations } from '@core/DataCook/Domain/ETL';

export abstract class JdbcPersistConfiguration extends ThirdPartyPersistConfiguration {
  abstract className: ThirdPartyPersistConfigurations;
  displayName: string;
  username: string;
  password: string;
  tableName: string;
  databaseName: string;
  persistType: PERSISTENT_TYPE;

  protected constructor(displayName: string, username: string, password: string, databaseName: string, tableName: string, persistType: PERSISTENT_TYPE) {
    super();
    this.displayName = displayName;
    this.username = username;
    this.password = password;
    this.databaseName = databaseName;
    this.tableName = tableName;
    this.persistType = persistType;
  }

  getId(): string {
    return [this.className, this.databaseName, this.tableName].join('_');
  }
}

File: schema-service/src/main/scala/datainsider/data_cook/domain/persist/JdbcPersistConfiguration.scala
package datainsider.data_cook.domain.persist

import com.fasterxml.jackson.module.scala.JsonScalaEnumeration
import datainsider.data_cook.domain.persist.PersistentType.PersistentType

@deprecated("use JDBCPersistOperator instead")
abstract class JdbcPersistConfiguration() extends ThirdPartyPersistConfiguration {
  def tableName: String
  def databaseName: String
  def persistType: PersistentType
  val displayName: Option[String] = None

  override def validate(): Unit = Unit
}

case class OracleJdbcPersistConfiguration(
    host: String,
    port: Int,
    serviceName: String,
    username: String,
    password: String,
    databaseName: String,
    tableName: String,
    @JsonScalaEnumeration(classOf[PersistentTypeRef])
    override val persistType: PersistentType,
    sslConfiguration: Option[SSLConfiguration] = None,
    sslServerCertDn: String = "",
    retryCount: Int = 5,
    retryDelay: Int = 2,
    extraPropertiesAsJson: Option[String] = None
) extends JdbcPersistConfiguration

case class MySQLJdbcPersistConfiguration(
    host: String,
    port: Int,
    username: String,
    password: String,
    databaseName: String,
    tableName: String,
    @JsonScalaEnumeration(classOf[PersistentTypeRef])
    override val persistType: PersistentType,
    extraPropertiesAsJson: Option[String] = None
) extends JdbcPersistConfiguration

case class MsSQLJdbcPersistConfiguration(
    host: String,
    port: Int,
    username: String,
    password: String,
    // database name
    catalogName: String,
    // table schema name
    databaseName: String,
    tableName: String,
    @JsonScalaEnumeration(classOf[PersistentTypeRef])
    override val persistType: PersistentType,
    extraPropertiesAsJson: Option[String] = None
) extends JdbcPersistConfiguration

case class PostgresJdbcPersistConfiguration(
    host: String,
    port: Int,
    username: String,
    password: String,
    // mapping with database
    catalogName: String,
    // mapping with schema name
    databaseName: String,
    tableName: String,
    @JsonScalaEnumeration(classOf[PersistentTypeRef])
    override val persistType: PersistentType,
    extraPropertiesAsJson: Option[String] = None
) extends JdbcPersistConfiguration
case class VerticaPersistConfiguration(
    host: String,
    port: Int,
    username: String,
    password: String,
    catalog: String = "",
    databaseName: String,
    tableName: String,
    @JsonScalaEnumeration(classOf[PersistentTypeRef])
    override val persistType: PersistentType,
    sslConfiguration: Option[SSLConfiguration] = None,
    isLoadBalance: Boolean = false,
    extraPropertiesAsJson: Option[String] = None
) extends JdbcPersistConfiguration
